Update WorkflowInstanceDetails with ActivityExecutionRecord

sfmskywalker opened the pull request:
Refactored WorkflowInstanceDetails in Elsa.Studio.Workflows to use ActivityExecutionRecord instead of ActivityExecutionContextState. This involved changing variable references, adjusting async methods, and switching from a package reference to a project reference for Elsa.Api.Client in Elsa.Studio.Core.

sfmskywalker committed:
Update WorkflowInstanceDetails with ActivityExecutionRecord

Refactored WorkflowInstanceDetails in Elsa.Studio.Workflows to use ActivityExecutionRecord instead of ActivityExecutionContextState. This involved changing variable references, adjusting async methods, and switching from a package reference to a project reference for Elsa.Api.Client in Elsa.Studio.Core.

sfmskywalker committed:
Refactor WorkflowInstanceDetails in Elsa Studio

Removed commented out code and unnecessary default initializations in WorkflowInstanceDetails.razor.cs. This includes the commented out _workflowActivityExecutionContext and unnecessary default initializations for SelectedSubWorkflow and SelectedSubWorkflowExecutions. The refactoring increases code readability and efficiency.

MariusVuscanNx commented on src/framework/Elsa.Studio.Core/Elsa.Studio.Core.csproj:
```diff
@@ -33,12 +33,12 @@
-
     <ItemGroup Label="Elsa">
-        <PackageReference Include="Elsa.Api.Client"/>
+        <ProjectReference Include="..\..\..\..\..\elsa-core\main\src\clients\Elsa.Api.Client\Elsa.Api.Client.csproj"/>
```
How is this going to work?

sfmskywalker replied:
It won't; I am waiting for the elsa-core PR to be completed & built, so that I can update this solution to point to a newer version of Elsa.Api.Client.

sfmskywalker committed:
Update Elsa.Api.Client version and reference method

The Elsa.Api.Client version has been updated from 3.1.0-preview.1152 to 3.1.0-preview.1163 in the Directory.Packages.props file. Additionally, the reference method to the Elsa.Api.Client in the Elsa.Studio.Core project has been changed from a direct project reference to a package reference.
